gincmd/uploadcmd.go

```diff
@@ -34,15 +35,27 @@ func upload(cmd *cobra.Command, args []string) {
 			break
 		}
 	}
-
+	fmt.Println(args)
 	paths := args
 	if len(paths) > 0 {
 		commit(cmd, paths)
 	}
 
 	uploadchan := make(chan git.RepoFileStatus)
 	go gincl.Upload(paths, remotes, uploadchan)
-	formatOutput(uploadchan, 0, jsonout)
+	if verbose == true {
+		var cmd_spec_var []string
+		path, _ := git.RemoteShow()
+		for _, v := range path {
```

Comment:
Why loop twice over the paths?
```go
for _, v := range path {
```
gives you the URL anyway, so you can just use it for the print instead of collecting them in a string slice and then looping again.

Generally though, I don't know if this block is necessary.  I think the raw output would be enough (I'll have to check) but even so, this seems to be printing all the remotes before starting the `formatOutput` function, so it's not really doing anything useful.  It even prints all the remotes if only some of them were specified, right?